src/main/java/org/springframework/cloud/bindings/boot/BindingFlattenedEnvironmentPostProcessor.java

```diff
@@ -17,7 +17,7 @@
 package org.springframework.cloud.bindings.boot;
 
 import org.springframework.boot.SpringApplication;
-import org.springframework.boot.context.config.ConfigFileApplicationListener;
+import org.springframework.boot.context.config.ConfigDataEnvironmentPostProcessor;
```

Comment:
@pivotal-david-osullivan : this class, `ConfigFileApplicationListener` was deprecated with [Spring Boot 2.4.0](https://github.com/spring-projects/spring-boot/blob/v2.4.0/spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/config/ConfigFileApplicationListener.java#L115) and disappeared with Spring Boot 3.0.0.
I was afraid it would break compatibility, but since this order is just here for a static int, it seems like whether we use the old class or new one (`ConfigDataEnvironmentPostProcessor`), that's fine!
All in all, I believe if we keep building SCB with Java 1.8 (and hence, Spring Boot <3), we shall not break consumers using Spring Boot 2.x , and still be compatible with Spring Boot 3 (as long as we carefully select the appropriate properties according to Spring Boot 2 or 3 like it was suggested)